src/cryptoadvance/specter/templates/wallet/send/new/wallet_send.jinja

```diff
@@ -501,35 +468,39 @@
 		}
 
 		function validateForm(submitted=false) {
+			const recipients = getRecipients();
+			// this assures that the 'recipient_dicts' input has the newest recipientDicts
+			document.getElementById('recipient_dicts').value  = JSON.stringify(getRecipientDicts());  
 			console.log("validateForm is called")
 			let createPSBTButton = document.getElementById('create_psbt_btn');
 			// Disables submit but button still clickable before every validation
 			createPSBTButton.setAttribute('type', 'button');
 			let totalAmount = 0.0;
 			let assetAmounts = {};
 			if (document.getElementById('ui-radio-btn').checked) {
-				for(let i in amounts) {
-					let unit = units[i]
-					let amount = amounts[i]
-					if(unit == 'btc' || unit == 'sat'){
-						totalAmount += (unit == 'sat' ? amount / 1e8 : amount);
+				for(let i in recipients) {
+					let recipient = recipients[i];
+					if(recipient.unit == 'btc' || recipient.unit == 'sat'){
+						totalAmount += (recipient.unit == 'sat' ? recipient.amount / 1e8 : recipient.amount);
 					}else{
-						if(unit in assetAmounts){
-							assetAmounts[unit] += amount;
+						if(recipient.unit in assetAmounts){
+							assetAmounts[recipient.unit] += recipient.amount;
 						}else{
-							assetAmounts[unit] = amount;
+							assetAmounts[recipient.unit] = recipient.amount;
 						}
 					}
-					if (document.getElementById("amount_" + i).value == '') {
-						amount = 0;
+					if (document.getElementById(`recipient_${recipient.recipientId}`).shadowRoot.getElementById('amount').value == '') {
```

Comment:
I would remove this, it is better feedback to the validation fail without setting the amount to zero beforehand.

Reply:
Ok.